src/backend/access/heap/rewriteheap.c

```diff
@@ -804,7 +804,7 @@ wallog_mapping_file(char const* path, int fd)
 	{
 		off_t size = lseek(fd, 0, SEEK_END);
 		char* buf;
-		elog(DEBUG1, "neon: writing contents of rewrite file %s, size %ld", path, size);
+		elog(DEBUG1, "neon: writing contents of rewrite file %s, size %ld", path, (long)size);
```

Comment:
What do you think about changing the format specifier to %jd?

Reply:
I am not sure if it is supported at all platforms.
IMHO explicit cast to long is safer:)

Reply:
At least there are no places in Postgres where "%rd" is used and more than 100 places where explicit cast to long `(long)` is used together with `"%ld"` format.

Reply:
Sounds good, though I would say lld and long long are better since long on windows is 32 bit if my memory is correct

Reply:
I do not think that it is somehow critical:
- it is debug message
- mapping files never can be larger than 1Gb
- we are not going to run Neon at Windows